specification/schemas/metadata_schema_v0.2.json

```diff
@@ -0,0 +1,49 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "id": "metadata_schema_v0.2.json",
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string",
+            "regex": "[a-z0-9]([-a-z0-9]*[a-z0-9])?"
+        },
+        "version": {
+            "type": "string"
+        },
+        "description": {
+            "type": "string"
+        },
+        "namespace": {
+            "type": "string"
+        },
+        "maintainers": {
+            "type": "array",
+            "items": {
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "email": {
+                        "type": [
+                            "string",
+                            "null"
+                        ]
+                    }
+                }
+            }
+        },
+        "targets": {
+            "properties": {
+                "swarm": {
+                    "type": "boolean"
+                },
+                "kubernetes": {
+                    "type": "boolean"
+                }
+            }
+        }
+    },
+    "required": [
+        "name"
```

Comment:
Should version be required as well? I'm thinking about how we use this in the client to tag the images. I feel like probably yes.

Reply:
Sure! And what about the description? With this schema, description can't be null (so the `docker-app init` result is not valid against this schema). One solution is to modify the type `string` to `string or null` type, the other is to modify the `init` command to add an empty "" string to the description.

Reply:
Personally I think `string or null` is slightly nicer here. Semantically I think "" and blank are the same in this context. But happy to go with others opinions.

Reply:
👍